src/kOS/Suffixed/ResourceTransferValue.cs

```diff
@@ -0,0 +1,387 @@
+                    if (resource == null) continue;
+
+                    var transferAmount = Math.Min(remaining, evenShare);
+
+                    SafeHouse.Logger.Log(string.Format("TRANSFER WORK: {0} PUT AMOUNT: {1}", part.flightID, transferAmount));
+
+                    remaining += part.TransferResource(resource.info.id, transferAmount);
+
+                    SafeHouse.Logger.Log(string.Format("TRANSFER WORK: {0} PUT REMAINING: {1}", part.flightID, remaining));
+                }
+                retries++;
+            }
+        }
+
+        /// <summary>
+        /// Requests the resource from all parts in the collection
+        /// </summary>
+        /// <param name="parts">All of the donor parts</param>
+        /// <param name="transferGoal">the aggregate amount we are seeking to remove from parts</param>
+        /// <returns>the amount we were successful at pulling</returns>
+        private double PullResources(IList<global::Part> parts, double transferGoal)
+        {
+            double toReturn = 0.0;
+            var availableResources = CalculateAvailableResource(parts);
+            foreach (var part in parts)
+            {
+                var resource = part.Resources.Get(resourceInfo.id);
+                if (resource == null) continue;
+
+                var thisPartsPercentage = resource.amount/availableResources;
+
+                // Throttle the transfer
+                var thisPartsShare = transferGoal*thisPartsPercentage;
+                var thisPartsRate = resource.maxAmount*RESOURCE_SHARE_PER_UPDATE;
+                
+                // The amount you pull must be negative 
+                thisPartsShare = -Math.Min(thisPartsShare, thisPartsRate);
+                // the amount is subject to floating point lameness, if we round it here it is not material to the request but should make the numbers work out nicer.
+                thisPartsShare = Math.Round(thisPartsShare, 5);
+
+                SafeHouse.Logger.Log(string.Format("TRANSFER WORK: {0} PULL AMOUNT: {1}", part.flightID, thisPartsShare));
+
+                toReturn += part.TransferResource(resourceInfo.id, thisPartsShare);
+
+                SafeHouse.Logger.Log(string.Format("TRANSFER WORK: {0} PULLED RESOURCE: {1}", part.flightID, toReturn));
+            }
+            SafeHouse.Logger.Log("TRANSFER WORK: Pulled: " + toReturn);
+            return toReturn;
+        }
+
+        private double CalculateTransferGoal(IEnumerable<global::Part> toParts)
+        {
+            double toReturn;
+
+            var destinationAvailableCapacity = CalculateAvailableSpace(toParts);
+
+            if (!amount.HasValue)
+            {
+                toReturn = destinationAvailableCapacity;
+            }
+            else
+            {
+                var rawGoal = amount.Value - transferedAmount; 
+                toReturn = Math.Min(destinationAvailableCapacity, rawGoal);
+            }
```

Comment:
I completely agree. Like you said, its pretty easy to miss and they should be swapped. Thanks for catching that